File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        string ans = "";
        string temp = "";
        for(char curr:s){
            if(curr == ' '){
                if(temp!=""){
                    ans = temp+" "+ans;
                    temp = "";
                }
            }else{
                temp+=curr;
            }
        }
        if(temp!=""){
                    ans = temp+" "+ans;
                    
                }
        return ans.substr(0,ans.size()-1);
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" ");
        StringBuilder builder = new StringBuilder();
        for(int i = words.length-1;i>=0;i--){
            if(!words[i].equals("")){
                builder.append(words[i]);
                builder.append(" ");
            }
        }
        String buildString = builder.toString();
        buildString = buildString.substring(0,buildString.length()-1);
        return buildString;
    }
}